A58-client-side-weighted-round-robin-lb-policy.md

```diff
@@ -0,0 +1,251 @@
+
+#### Java
+`OrcaPerRequestUtil.OrcaReportingTracerFactory` is available in `io.grpc.xds.orca` to process per-call backend load reports in `ClusterImplLoadBalancer`. This needs to be moved to a new package and shared with `weighted_round_robin`. Then the picker attaches this tracer to the subchannel selection and the listener takes the reference to the subchannel to which it updates the weight to in `onLoadReport()`.
+
+#### Go
+Per-call load report handling is fully implemented. Picker selection result `PickResult` has an optional callback `Done()` that is passed with the backend load metrics parsed from the response if available. With `weighted_round_robin`, the picker only sets Done() in `PickResult` when per-call load reporting is enabled, and the callback updates the weight.
+
+### OOB Load Reporting
+
+OOB load report handling in clients depends on [gRFC A51]. This section outlines how this feature works with `weighted_round_robin`.
+
+In all languages, `report_interval` in `OrcaLoadReportRequest` is determined by the service config parameter `oob_reporting_period`. When it changes, a new stream request is sent. In Java/Go, the implementation needs to add support to enable and disable the OOB listener when `enable_oob_load_report` is updated. In C++ this happens automatically as a new subchannel list is created.
+
+`weighted_round_robin` initiates OOB load reporting when the LB policy parameter `enable_oob_load_report` changes to true. When the backend returns `UNIMPLEMENTED` to the OOB load report request it proceeds with a log message as if it is not enabled and does not retry.
+
+#### C/C++
+
+Processing OOB load reports is supported with `OobBackendMetricWatcher`. `weighted_round_robin` implements this interface. The watcher takes the reference of the subchannel weight object and update weights to it.
+
+#### Java
+
+`OrcaOobUtil` is available in `io.grpc.xds.orca` but currently not in use. This needs to be moved to a new package and shared with `weighted_round_robin`, which implements `ForwardingLoadBalancerHelper` that takes the helper from `OrcaOobUtil`. The listener takes the reference to the subchannel to which it updates the weight to in `onLoadReport()`. A new API is needed to set and unset the listener in `OrcaOobUtil` to toggle the OOB load report processing.
+
+#### Go
+
+`orca.OOBListener` is available for OOB load reporting. `weighted_round_robin` extends it and attach it to the subchannel to which it updates the weight in `onLoadReport()`. A new API is needed to support enabling and disabling OOB load reporting.
+
+
+### xDS Integration
+
+`weighted_round_robin` is enabled as the endpoint picking policy of `WrrLocality` of [gRFC A52]. The implementation is available in C++ and Java but not in Go yet.
```

Comment:
Please remove the second sentence.  This gRFC is about the design, not the implementation timeline.

Reply:
Done.